### ISSUE TITLE
ntp service is checked even chrony is installed and configured

### ISSUE THREAD

**Issue by JugeHuge** [ovh/debian-cis#192]
If chrony or systemd-timesyncd are used, ntp should be removed.
Only one time synchronization method should be in use on the system.

**Comment by ThibaultDewailly** [comment]
Hello,

I'm not sure what you mean here, regarding audit if either one of the three packages is installed (chrony, systemd-timesyncd or ntp) then 2.2.1.1_use_time_sync.sh is ok.
you can then enable the hardened configuration in 2.2.1.2_configure_systemd-timesyncd.sh 2.2.1.3_configure_chrony.sh or 2.2.1.4_configure_ntp.sh respectively.

You want to purge other package during the apply phase in the 2.2.1.1_use_time_sync.sh script ?
If so, you can proceed and patch the apply function to suit your need.

**Comment by JugeHuge** [comment]
Hmm. Can't really remember what did i meant with this. :D Have to test it again.
If i remember correctly i had chrony installed and configured but still it tried to check if ntp is configured and hence gave failure from it with audit option. I'll check it and get back to this.

**Comment by JugeHuge** [comment]
Yes. found logs. So Chrony is installed and its audit goes ok but ntp audit fails with unexpected exit code.

2.2.1.4_configure_ntp     [INFO] Working on 2.2.1.4_configure_ntp
2.2.1.4_configure_ntp     [INFO] [DESCRIPTION] Configure Network Time Protocol (ntp). Check restrict parameters and ntp daemon runs ad unprivileged user.
2.2.1.4_configure_ntp     [INFO] Checking Configuration
2.2.1.4_configure_ntp     [WARN] ntp is not installed, not handling configuration
2.2.1.4_configure_ntp     [ KO ] Check failed with unexpected exit code: 2

**Comment by JugeHuge** [comment]
Well isn't that giving false audit result as you have Chrony installed and ntp package removed which then fails on check.  Should audit check that there is atleast one time sync service installed and not check all three as individually which distort audit result?

**Comment by ThibaultDewailly** [comment]
If I understand you correctly, and if I sum up my thought :

https://github.com/ovh/debian-cis/blob/master/bin/hardening/2.2.1.1_use_time_sync.sh#L20 verifies "Should audit check that there is atleast one time sync service installed"
"not check all three as individually which distort audit result" => pick only one of the three, the one you have chosen, and the audit is thorough and precise

**Comment by JugeHuge** [comment]
Ok so there was fix already coming. Have to test that.
Have to be more precise next time when writing issue. 😄 Thanks.
